pkg/queue/priority_queue_test.go

```diff
@@ -0,0 +1,142 @@
+package queue
+
+import (
+	"math/rand"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
```

Comment:
testify is not really used in this library

Reply:
`go.sum` is not changed, meaning gobox has an indirect dependency on `testify` already. I see [gotest.tools/v3/assert](https://github.com/gotestyourself/gotest.tools) is being used, but given [testify](https://github.com/stretchr/testify) is more popular and richer than `gotest.tools/v3/assert`, what is the problem of using `testify`?

Reply:
testify is indirectly referenced but if you look the entire rest of the unit tests in this file contain no testify-based assertions.  You have to adapt to the repo, and cannot just introduce a new testing paradigm or strengthen ties to an indirect dependency.

Reply:
I don't think sticking with `gotest` is a good idea. Look at some of the existing gobox test codes. Some are creating assertion helpers (DeepEqual, InDelta,  WithinDuration) which are not reusable across packages, some are creating suite and before/after test. All of these are available in `testify` already. I think it is better to move to a better testing library rather than writing repeated test helpers with `gotest` that are not reusable across packages.

* TestSuite
	* [gotest](https://github.com/getoutreach/gobox/blob/v1.54.0/pkg/log/fatal_test.go#L17)
	* [testify suite](https://pkg.go.dev/github.com/stretchr/testify@v1.8.0/suite)
* Before/AfterTest
	* [gotest](https://github.com/getoutreach/gobox/blob/v1.54.0/pkg/cli/github/auth_test.go#L22-L23)
	* [testify](https://pkg.go.dev/github.com/stretchr/testify@v1.8.0/suite#BeforeTest)
* DeepEqual
	* [gotest](https://github.com/getoutreach/gobox/blob/v1.54.0/pkg/log/caller_test.go#L31)
	* [testify EquvalValues](https://pkg.go.dev/github.com/stretchr/testify@v1.8.0/assert#EqualValues)
* Delta
	* [gotest](https://github.com/getoutreach/gobox/blob/v1.54.0/pkg/async/pool/utils_test.go#L10-L34)
	* [testify](https://pkg.go.dev/github.com/stretchr/testify@v1.8.0/assert#Assertions.InDelta)

Reply:
Moving this conversation to [here](https://outreach-hq.slack.com/archives/C01CAR6GVPD/p1664315615985359).